### PR TITLE
Update the `sub` claim

### PR DESCRIPTION
**Pull request by filiptronicek**
## Description

Adds the organization and user ID along with the context URL to the subject claim of our OIDC JWT token.

## Related Issue(s)

Fixes IDE-103

## How to test

In the preview environment, create a workspace and execute `gp idp token --audience gitpod.io` within it. After that, inspect the outputted token with a tool like https://jwt.io to verify it has all the fields this PR promises.

If you're not into testing yourself, I did it for you already. Here's the decoded token:
```json
{
  "iss": "https://api.ft-sub-update.preview.gitpod-dev.com/idp",
  "aud": [
    "gitpod.io"
  ],
  "azp": "gitpod.io",
  "c_hash": "MbIWgwnQBOAorEi9dN1-UQ",
  "exp": 1687627102,
  "iat": 1687623502,
  "auth_time": 1687623502,
  "email": "filip@gitpod.io",
  "name": "filiptronicek",
  "org_id": "6730b752-6dd6-48d0-9bc1-e8ac168d0f13",
  "sub": "org_id:6730b752-6dd6-48d0-9bc1-e8ac168d0f13:user_id:9a12464b-eb40-4879-8437-710fcb585441:context:https%3A%2F%2Fgithub.com%2Fcsweichel%2Foci-tool"
}
```

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - ft-sub-update</li>
	<li><b>🔗 URL</b> - <a href="https://ft-sub-update.preview.gitpod-dev.com/workspaces" target="_blank">ft-sub-update.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>

/hold

### CODE REVIEW

File: components/public-api-server/pkg/apiv1/identityprovider.go [hunk 1]
```diff
@@ -9,6 +9,7 @@ import (
 	"fmt"
 
 	connect "github.com/bufbuild/connect-go"
+	"github.com/filiptronicek/subber"
```
Comment (easyCZ) [inline components/public-api-server/pkg/apiv1/identityprovider.go:12]: Do we really need a dependency to a personal project here?

I'm not convinced the extra dependency is really bringing any value here.

Comment (filiptronicek) [reply components/public-api-server/pkg/apiv1/identityprovider.go:12]: I'm fine with copying the code over, let's do it.

**Comment by loujaybee** [comment]
@filiptronicek -> Just about to send a message internally about this PR, can we hold off from merging for now?

File: components/public-api-server/pkg/apiv1/identityprovider.go (hunk 1 above)
Comment (easyCZ) [reply components/public-api-server/pkg/apiv1/identityprovider.go:12]: Given it's just a single function really, let's include it here. Also helps prevent a `leftpad` scenario

**Comment by filiptronicek** [comment]
> can we hold off from merging for now

totally

File: components/public-api-server/pkg/apiv1/identityprovider.go (hunk 1 above)
Comment (filiptronicek) [reply components/public-api-server/pkg/apiv1/identityprovider.go:12]: @easyCZ are we ok with having the code in this file or do we move it somewhere else?

File: components/public-api-server/pkg/apiv1/identityprovider.go [hunk 2]
```diff
@@ -84,7 +84,18 @@ func (srv *IdentityProviderService) GetIDToken(ctx context.Context, req *connect
...
+	fields := []Claim{
+		{Key: "org_id", Value: workspace.Workspace.OrganizationId},
+		{Key: "user_id", Value: user.ID},
+		{Key: "context", Value: workspace.Workspace.ContextURL},
+	}
+
+	subject, err := s.PrepareSubClaim(fields...)
+	if err != nil {
+		log.Fatalf("Error preparing claim: %v", err)
+	}
```
[the remark this answers is not in the document]
Comment (filiptronicek) [reply components/public-api-server/pkg/apiv1/identityprovider.go:97]: https://github.com/gitpod-io/gitpod/pull/18045/commits/1561faa70cb8df469b61152c656cc1e2fdece489

**Comment by easyCZ** [comment]
Will come back to the review, once we're ready proceed with the change (aka, not in draft)

**Pull request closed without merging**